### ISSUE TITLE
can not decode.

### ISSUE THREAD

**Issue by yonghumeijj** [mikerabat/DelphiCBOR#3]
[data.zip](https://github.com/mikerabat/DelphiCBOR/files/14812012/data.zip)

This Data can not  decode. with 

Project Project1.exe raised exception class ECBorNotImplmented with message 'Error code $D9 + not implemented'.

**Comment by mikerabat** [comment]
It has been a while… but correct me if I‘m wring this is kinda of a Tag element for another data entity right? (Optional things are not yet in the lib…)I‘ll check what I can do about that… Kind regardsMikeVon meinem iPhone gesendetAm 30.03.2024 um 13:46 schrieb 小宝软件 ***@***.***>:﻿
data.zip
This Data can not  decode. with
Project Project1.exe raised exception class ECBorNotImplmented with message 'Error code $D9 + not implemented'.

**Comment by mikerabat** [comment]
Hi!

I was able to take a look at the file ... finally...

So the base problem is that the file starts with $D9D9F7 . This byte
sequence is defined to be just an indicator that the following
stream is actually encoded as CBOR. The decoder I implemented actually does
not look at such a header so the exception atcually
makes sense since it is not part of the decoding process in a stream but
rather an indicater that after that tag the cbor stream starts.

I'm still not sure if I shall implement a check for that... in the RFC it
is indicated that this could e.g. be part of a combined JSON CBOR
encoder and the value is chosen to not be part of any unicode or similar
decoding.....

Eventually I will add another parameter in the decode routine to allow that
header value so decoding succeeds if that value is there but still
there is a stricter decoder for other applications...

kind regards
   Mike
